internal/services/subscription/subscription_locations_data_source.go

```diff
@@ -0,0 +1,113 @@
+package subscription
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/resources/mgmt/2021-01-01/subscriptions"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/clients"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/subscription/parse"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/timeouts"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+func dataSourceSubscriptionLocations() *pluginsdk.Resource {
+	return &pluginsdk.Resource{
+		Read: dataSourceSubscriptionLocationsRead,
+
+		Timeouts: &pluginsdk.ResourceTimeout{
+			Read: pluginsdk.DefaultTimeout(5 * time.Minute),
+		},
+
+		Schema: map[string]*pluginsdk.Schema{
+			"subscription_id": {
+				Type:         pluginsdk.TypeString,
+				Required:     true,
+				ValidateFunc: validation.IsUUID,
+			},
+
+			"locations": {
+				Type:     pluginsdk.TypeSet,
```

Comment:
.. which Set's aren't usable in a Data Source as everything gets the same hash code - allows the `extended_locations` field to be turned into a (sorted) list as a top-level field

Reply:
Updated to "TypeList"